src/Evolution/Systems/Cce/BoundaryData.hpp

```diff
@@ -221,4 +333,218 @@ void dyads(
     gsl::not_null<tnsr::I<ComplexDataVector, 2, Frame::RadialNull>*>
         up_dyad) noexcept;
 
+/*!
+ * \brief Compute the \f$\beta\f$ (lapse) function for the CCE Bondi-like
+ * metric.
+ *
+ * \details The Bondi-like metric has \f$g^{u r} = - e^{2 \beta}\f$, and the
+ * value of \f$\beta\f$ is obtained from the intermediate null metric by (see
+ * equation (51) of \cite Barkett2019uae) using:
+ *
+ * \f[
+ * \beta = -\frac{1}{2} \ln r_{, \lamdba}
+ * \f]
+ */
+void beta_worldtube_data(
+    gsl::not_null<Scalar<SpinWeighted<ComplexDataVector, 0>>*> beta,
+    const tnsr::a<DataVector, 3, Frame::RadialNull>& d_bondi_r) noexcept;
+
+/*!
+ * \brief Compute the \f$U\f$ (shift) function for the CCE Bondi-like metric.
+ *
+ * \details The Bondi-like metric has \f$g^{r A} = -e^{-2 \beta} U^A\f$, and the
+ * spin-weighted vector \f$U = U^A q_A\f$. The value of \f$U^A\f$ can be
+ * computed from the intermediate null metric quantities (see equation (54) of
+ * \cite Barkett2019uae) using:
+ *
+ * \f[
+ * U = -( r_{,\lambda} g^{\lambda A} + r_{,B} g^{A B}) q_A / r_{,\lambda}
+ * \f]
+ *
+ */
+void bondi_u_worldtube_data(
+    gsl::not_null<Scalar<SpinWeighted<ComplexDataVector, 1>>*> bondi_u,
+    const tnsr::i<ComplexDataVector, 2, Frame::RadialNull>& down_dyad,
+    const tnsr::a<DataVector, 3, Frame::RadialNull>& d_bondi_r,
+    const tnsr::AA<DataVector, 3, Frame::RadialNull>&
+        inverse_null_metric) noexcept;
+
+/*!
+ * \brief Compute the \f$W\f$ (mass aspect) function for the CCE Bondi-like
+ * metric.
+ *
+ * \details The Bondi-like metric has \f$g^{rr} = e^{-2 \beta}(1 + r W)\f$. The
```

Comment:
g or gamma?

Reply:
The up-index spatial quantities here are correct as g. The down-index ones could really be written as either, but the literature for Bondi-like metric and radial null metric do not tend to perform a 3+1 split, so the references will be easier to compare if we just stick with g's

Reply:
I'm now worried that all this code is filled with incorrect information... Are you telling me that anytime you call `tnsr::II` the inverse spatial metric it's not?

Reply:
Sorry, I should be more specific. There are effectively three frames in this computation:
1) The Cauchy (`Inertial`) frame, from the input data, performs a 3+1 split on the metric. This is the only one for which any `tnsr::ii` or `tnsr:II`s appear.
2) The intermediate null (`RadialNull`) frame, in which the metric is expressed but only ever as a `tnsr::AA` or `tnsr::aa` spacetime metric.
3) The Bondi-like metric, which does not have a frame declaration of its own, and instead is directly expressed in terms of the Bondi scalars \beta, W, U, J, and H without going through a tensor quantity with indices at all.

I do admit that this computation complicated, and perhaps insufficiently explained in the dox. Do you have a suggestion for a good place to put the above information? Or is it crucial enough for understanding to be duplicated on most of these free functions?

Reply:
Ah okay! That makes sense. Thank you for clarifying :)

I think the best place will be the CCE namespace once you're ready to document the full CCE procedure. I think we discussed this on a previous PR and you said that'll come in the future so as long as you're willing to enlighten me along the way I think holding off and documenting the overall procedure either in a tutorial or as part of the namespace is the right thing to do. I don't think there's a need to have that info on each of these functions :)